### PR TITLE
Add Q4 roadmap items

### PR DESCRIPTION
**Pull request by daviddias**
Here is the q4 roadmap stub. Let's go through each of these milestones in the next Monday call and complete them with the tasks that need to be performed.

//cc @jbenet @flyingzumwalt @em-ly 

@whyrusleeping @lgierth @Kubuxu could you check and add the milestones that you are owning?

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -123,21 +123,24 @@ Having the community guidelines pointed to, and creating custom ones as needed,
...
 ## Milestone - libp2p.io
 
-> Summary:
+> Summary: We need to ship a first version of the libp2p.io website
 
 ### Leads
 
-- David
-- Richard
+- @VictorBjelkholm
```
Comment (daviddias) [inline ROADMAP.md:121]: You feel like taking ownership of the content as well?

File: ROADMAP.md [hunk 2]
```diff
@@ -123,21 +123,24 @@ Having the community guidelines pointed to, and creating custom ones as needed,
...
 
 ### Tasks
 
-- [ ] libp2p.io - https://github.com/libp2p/libp2p-website
+- [ ] Agree and create tasks about what's missing for first version
+- [ ] Deploy and setup CD for shipping website
```
Comment (daviddias) [inline ROADMAP.md:130]: Some of the tasks that are assigned to me are:
- Come up with the sitemap
- Libp2p examples
- Set what is the inspiration (pretty much D3.js and leveldb.org website)
- Fill the website with content

File: ROADMAP.md (hunk 1 above)
Comment (victorb) [reply ROADMAP.md:121]: Absolutely, I'll take ownership of the whole thing, making sure we have a first iteration of the website that is actually usable, which includes content :)

File: ROADMAP.md [hunk 3]
```diff
@@ -2,23 +2,28 @@
...
-- [Milestone 7 - Break DHT into Peer Routing and Content Routing](#milestone-7---break-dht-into-peer-routing-and-content-routing)
-- [Milestone 8 - Packet Switching / Overlay Network](#milestone-8---packet-switching--overlay-network)
+- [Milestone - JS and Go libp2p interop](#milestone-js---and-go-libp2p-interop)
+- [Milestone - Improve Connectivity of go-libp2p/go-ipfs](#milestone---improve-connectivity-of-go-libp2pgo-ipfs)
+- [Milestone - Standardize Organization](#milestone---standardize-organization)
+- [Milestone - libp2p.io](#milestone---libp2pio)
+- [Milestone - Developer Experience](#milestone---developer-experience)
+- [Milestone - Move DHT to use IPRS](#milestone---move-dht-to-use-iprs)
+- [Milestone - Break DHT into Peer Routing and Content Routing](#milestone---break-dht-into-peer-routing-and-content-routing)
+- [Milestone - Packet Switching / Overlay Network](#milestone---packet-switching--overlay-network)
```
Comment (whyrusleeping) [inline ROADMAP.md]: These last 3 arent Q4, right?

File: ROADMAP.md [hunk 4]
```diff
@@ -27,107 +32,203 @@
...
-  - [ ] Implement Relay in go-libp2p
-  - [ ] Technical Discussion: Improve NAT Traversal
-    - https://github.com/ipfs/notes/issues/110
-    - https://github.com/ipfs/go-ipfs/issues/2509
-  - [ ] Improve NAT traversal in go-libp2p
-- Dependencies
-  - n/a
-- Requirements by other projects
-  - n/a
+## Milestone - interface-xtp
```
Comment (Kubuxu) [inline ROADMAP.md:53]: Wasn't xtp being bumped of Q4?

File: ROADMAP.md (hunk 3 above)
Comment (victorb) [reply ROADMAP.md]: @whyrusleeping packet switching is at least, scheduled for Week 10, December 19

File: ROADMAP.md (hunk 4 above)
Comment (daviddias) [reply ROADMAP.md:53]: This is just defining the interface

**Comment by daviddias** [comment]
Missing:
- [ ] @lgierth to add his milestones (see: https://public.etherpad-mozilla.org/p/ipfs-oct-24-libp2p)
- [x] @Kubuxu to add the Private Networks milestone
- [x] @whyrusleeping to review and add anything if missing

**Comment by daviddias** [comment]
@lgierth @Kubuxu @whyrusleeping ping 🛎 :)

File: ROADMAP.md [hunk 5]
```diff
@@ -27,107 +32,203 @@
...
-### Notes
+- [ ] Examples
+  - [ ] How to use js-libp2p (dial, multi transport, stream muxing, identify, protocol multiplexing)
+  - [ ] How to use go-libp2p (everything js-libp2p + DHT)
+- [ ] Spec
+  - [ ] Move spec to the libp2p org
+  - [ ] Revisit and improve
+- [ ] interface-libp2p
+  - [ ] Make js-libp2p-ipfs and js-libp2p-ipfs-browser use it
+  - [ ] Create js-libp2p, a base class that other libp2p builds can reuse
```
Comment (whyrusleeping) [inline ROADMAP.md:274]: Add:
- make go-libp2p have a top level, easy to use package
- add examples for each of the go-libp2p-\* repos

File: ROADMAP.md [hunk 6]
```diff
@@ -137,58 +238,73 @@ n/a
...
+
+- [ ] Multigram specced and implemented in Golang
+  - https://github.com/multiformats/multigram
 
 ### Notes
 
 The overlay network will not have any multi-hop routing for now.
 We only introduce the packet switch and continue being a direct single-hop network.
 
 ### Expected date of completion: `Q3`
```
Comment (whyrusleeping) [inline ROADMAP.md]: Q3 is the expected completion date for a few milestones here, that doesnt seem right

File: ROADMAP.md [hunk 7]
```diff
@@ -137,58 +238,73 @@ n/a
...
+### Tasks
+
+- [ ] Technical Discussion: Define a strategy for connection closing
+  - https://github.com/libp2p/go-libp2p/issues/45
+  - https://github.com/ipfs/go-ipfs/issues/3065
+  - https://github.com/ipfs/notes/issues/110
+- [ ] Implement the strategy for connection closing in go-libp2p
+- [ ] Technical Discussion: Definte a strategy for Relay (or multiple Relay)
+  - https://github.com/ipfs/notes/issues/110
+- [ ] Implement Relay in go-libp2p
```
Comment (whyrusleeping) [inline ROADMAP.md:347]: I don't know if we agreed to have relay in this milestone. What was the conclusion there?

Comment (whyrusleeping) [reply ROADMAP.md:347]: Better question, is this doc just milestones in general? or is there some sort of weighting for Q4?

File: ROADMAP.md (hunk 3 above)
Comment (daviddias) [reply ROADMAP.md]: @lgierth is working on it, not the implementation, but the spec

File: ROADMAP.md (hunk 6 above)
Comment (daviddias) [reply ROADMAP.md]: These were 'cached' milestones, things that we have written down before but weren't allocated to Q4.

File: ROADMAP.md (hunk 7 above)
Comment (daviddias) [reply ROADMAP.md:347]: same as above

**Pull request merged**